server/src/test/java/io/druid/server/coordinator/DruidCoordinatorBalancerTest.java

```diff
@@ -395,4 +400,269 @@ public void testRun2() throws IOException
+    cluster.put("normal", MinMaxPriorityQueue.orderedBy(DruidCoordinatorBalancerTester.percentUsedComparator)
+                                             .create(
+                                                 Arrays.asList(
+                                                     new ServerHolder(druidServer1, peon1),
+                                                     new ServerHolder(druidServer2, peon2)
+                                                 )
+                                             ));
+
+    cluster.put("extra", MinMaxPriorityQueue.orderedBy(DruidCoordinatorBalancerTester.percentUsedComparator)
+                                            .create(
+                                               Arrays.asList(
+                                                   new ServerHolder(druidServer3, peon3),
+                                                   new ServerHolder(druidServer4, peon4)
+                                               )
+                                            ));
+
+    DruidCoordinatorRuntimeParams params =
+        DruidCoordinatorRuntimeParams.newBuilder()
+                                     .withDruidCluster(
+                                         new DruidCluster(cluster)
+                                     )
+                                     .withLoadManagementPeons(
+                                         ImmutableMap.<String, LoadQueuePeon>of(
+                                             "1",
+                                             peon1,
+                                             "2",
+                                             peon2,
+                                             "3",
+                                             peon3,
+                                             "4",
+                                             peon4
+                                         )
+                                     )
+                                     .withAvailableSegments(segments.values())
+                                     .withDynamicConfigs(
+                                         new CoordinatorDynamicConfig.Builder().withMaxSegmentsToMove(
+                                             1
+                                         ).build()
+                                     )
+                                     .withBalancerStrategy(balancerStrategy)
+                                     .withBalancerReferenceTimestamp(new DateTime("2013-01-01"))
+                                     .build();
+
+    DruidCoordinatorBalancerTester balancerTester = new DruidCoordinatorBalancerTester(coordinator);
+    params = balancerTester.run(params);
+    Assert.assertTrue(params.getCoordinatorStats().getPerTierStats().get("movedCount").get("normal").get() == 1);
+    Assert.assertTrue(params.getCoordinatorStats().getPerTierStats().get("movedCount").get("extra").get() == 1);
+
+    exec.shutdown();
```

Comment:
(Optional) Not really important for unit tests but this is not a guaranteed shutdown.